### PR TITLE
Add team flag to fly command `abort-build`

### LINKED ISSUE #5215: Add --team to remaining fly commands
## Goal

Implement `--team` for remaining fly commands. `fly/integration` tests should be updated. Cases should be added to the `error_handling` file as well https://github.com/concourse/concourse/blob/587483424a47f3eebb2ed8e861687b60cccb9ec7/fly/integration/error_handling_test.go#L92

```bash
# use default team
fly -t ci <command> <command-options>
# use another team
fly -t ci <command> <command-options> --team other-team
```

- [ ] abort-build
- [ ] resources
- [ ] check-resource-type
- [ ] check-resource
- [ ] resource-versions
- [ ] watch
- [ ] checklist
- [ ] clear-task-cache
- [ ] execute
- [ ] expose-pipeline
- [ ] destroy-pipeline
- [ ] get-pipeline
- [ ] set-pipeline
- [ ] hide-pipeline
- [ ] pause-pipeline
- [ ] unpause-pipeline
- [ ] rename-pipeline
- [ ] order-pipelines
- [x] ~jobs~
- [x] ~trigger-job~
- [x] ~pause-job~
- [x] ~unpause-job~
- [x] ~hijack~

## Background
Follow-up story from #4196. You can look at `trigger-job` for guidance on how to implement the flag.

https://github.com/concourse/concourse/blob/587483424a47f3eebb2ed8e861687b60cccb9ec7/fly/commands/trigger_job.go#L40-L47

### PR DESCRIPTION
**Pull request by EstebanFS**
## Changes proposed by this PR

closes various items in #5215

* [x] Small refactor in order to centralize find team logic
* [x] Add team flag to abort-build command (with the new refactor)

## Notes to reviewer

Added team flag to fly command abort-build

Now you could add the `--team` flag to specify the team this action will be applied

## Release Note

* Added `team` flag to fly command abort-build, you could use this following the next format
`fly -t dev abort-build --job some-pipeline/some-job -b 2 --team test`

### CODE REVIEW

File: fly/integration/abort_build_test.go [hunk 1]
```diff
@@ -194,4 +195,79 @@ var _ = Describe("AbortBuild", func() {
 			}).By(2))
 		})
 	})
+
+	FContext("user is NOT targeting the same team the pipeline belongs to", func() {
```
Comment (taylorsilva) [inline fly/integration/abort_build_test.go:199]: Left a focus here :)
```suggestion
	Context("user is NOT targeting the same team the pipeline belongs to", func() {
```

File: fly/commands/internal/flaghelpers/team_flag.go [hunk 2]
```diff
@@ -35,3 +36,16 @@ func (flag *TeamFlag) Complete(match string) []flags.Completion {
 func (flag TeamFlag) Name() string {
 	return string(flag)
 }
+
+func (flag TeamFlag) LoadTeam(target rc.Target) (concourse.Team, error) {
+	teamFlag := string(flag)
+	team := target.Team()
+	var err error
+	if teamFlag != "" {
```
Comment (taylorsilva) [inline fly/commands/internal/flaghelpers/team_flag.go:44]: The `Name()` func above does the casting for you.
```suggestion
	team := target.Team()
	var err error
	if flag.Name() != "" {
```

**Review by taylorsilva — CHANGES_REQUESTED** [review]
Missed one command in your refactor, `jobs`: https://github.com/concourse/concourse/blob/7e04f7d75f0770441d7946958204b9b796b117fe/fly/commands/jobs.go#L18

Nice refactor though! Should be wayyy easier to add the rest now :)

**Comment by EstebanFS** [comment]
Hey @taylorsilva , I just solved your comments, thanks for reviewing it!

**Review by taylorsilva — APPROVED** [review]
LGTM

**Pull request merged**